From 6aaa1886b4ff95b42879664bf6cf5ff71be888a6 Mon Sep 17 00:00:00 2001
From: Zeyu Fu <zeyufu@microsoft.com>
Date: Thu, 05 Jun 2025 19:35:18 -0700
Subject: [PATCH] Check observer_ before use in RestartForHls

Added a null check for the observer_ WeakPtr before calling
OnHlsManifestDetected() in RestartForHls. Since observer_ is a
WeakPtr, it could be null if the MediaObserver is destroyed.
This prevents a potential crash when HLS media is detected but
the observer is no longer valid.

Change-Id: I42b576336a10aae5bd7a6234a8e047b8986e3d8b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6621101
Reviewed-by: Eugene Zemtsov <eugene@chromium.org>
Reviewed-by: Dale Curtis <dalecurtis@chromium.org>
Commit-Queue: Zeyu Fu <zeyufu@microsoft.com>
Cr-Commit-Position: refs/heads/main@{#1470327}
---

--- a/third_party/blink/renderer/platform/media/web_media_player_impl.cc
+++ b/third_party/blink/renderer/platform/media/web_media_player_impl.cc
@@ -1980,7 +1980,9 @@
 void WebMediaPlayerImpl::RestartForHls() {
   DCHECK(main_task_runner_->BelongsToCurrentThread());
 #if BUILDFLAG(ENABLE_HLS_DEMUXER)
-  observer_->OnHlsManifestDetected();
+  if (observer_) {
+    observer_->OnHlsManifestDetected();
+  }
   SetMemoryReportingState(false);
   StartPipeline();
 #else
